crowbar_framework/lib/crowbar/lock.rb

```diff
@@ -29,13 +29,14 @@ def initialize(options = {})
     end
 
     def self.with_lock(options = {})
-      new(options).with_lock do
+      shared = options.delete(:shared) || false
```

Comment:
why not `options.fetch(:shared, false)` ? we are using this in many places in crowbar btw